r/tools/nixlibs.R

```diff
@@ -320,33 +300,54 @@ build_libarrow <- function(src_dir, dst_dir) {
     BUILD_DIR = build_dir,
     DEST_DIR = dst_dir,
     CMAKE = cmake,
+    # EXTRA_CMAKE_FLAGS will often be "", but it's convenient later to have it defined
```

Comment:
Why?

Reply:
Later in the code I have:
https://github.com/apache/arrow/blob/98b5601f94ff0f0caf240c6e1b914d4e8e49f98e/r/tools/nixlibs.R#L447-L452

If we don't add `EXTRA_CMAKE_FLAGS` to the vector, that section could instead be
```r
    # The syntax to turn off XSIMD is different.
    # Pull existing value of EXTRA_CMAKE_FLAGS first (must be defined)
    "EXTRA_CMAKE_FLAGS" = paste(
      Sys.getenv("EXTRA_CMAKE_FLAGS"),
      "-DARROW_SIMD_LEVEL=NONE -DARROW_RUNTIME_SIMD_LEVEL=NONE"
    )
```

I did it the first way to have the `EXTRA_CMAKE_FLAGS` collected at the same time as the other existing build flags, but if you think the second way is cleaner, I'm happy to change it.